samples/adc_mux/main.cpp

```diff
@@ -0,0 +1,31 @@
+#include <EVT/io/GPIO.hpp>
+#include <EVT/io/manager.hpp>
+#include <EVT/dev/ADCMux.hpp>
+#include <EVT/utils/time.hpp>
+
+namespace IO = EVT::core::IO;
+namespace DEV = EVT::core::DEV;
+namespace time = EVT::core::time;
+
+int main() {
+    IO::init();
+
+    IO::GPIO& ctrlGPIO1 = IO::getGPIO<IO::Pin::PB_10>(IO::GPIO::Direction::OUTPUT);
+    IO::GPIO& ctrlGPIO2 = IO::getGPIO<IO::Pin::PB_4>(IO::GPIO::Direction::OUTPUT);
+
+    IO::ADC& adc = IO::getADC<IO::Pin::PB_8>();
+
+    DEV::ADCMux mux(ctrlGPIO1, ctrlGPIO2, adc);
+
+    IO::GPIO& ctrlGPIO3 = IO::getGPIO<IO::Pin::PB_8>(IO::GPIO::Direction::OUTPUT);
+
+    while(1) {
+        ctrlGPIO3.writePin(EVT::core::IO::GPIO::State::LOW);
+        mux.setOutputPin(mux.getInput());
```

Comment:
I see that you set the output pin based on the mux input. You should be cycling through valid inputs (0-3 in this case) and reading the adc output.`getInput` could return a large range of values based on the ADC and could thus be something not valid for the output select on the mux

Reply:
Would this be a check of if or statements for a valid input, ie. 0,1,2,3?

Reply:
You could test this either using the UART to get user input, or via some timing. For example cycle through 0 to 3 waiting for a couple of seconds between cycle of a loop